src/electionguard/guardian.py

```diff
@@ -384,3 +433,76 @@ def partially_decrypt(
         )
 
         return (partial_decryption, proof)
+
+    def compensate_decrypt(
+        self,
+        missing_guardian_id: str,
+        elgamal: ElGamalCiphertext,
+        extended_base_hash: ElementModQ,
+        nonce_seed: ElementModQ = None,
+        decrypt: AuxiliaryDecrypt = default_auxiliary_decrypt,
+    ) -> Optional[Tuple[ElementModP, ChaumPedersenProof]]:
+        """
+        Compute a compensated partial decryption of an elgamal encryption 
+        on behalf of the missing guardian
+
+        :param missing_guardian_id: the guardian 
+        :param elgamal: the `ElGamalCiphertext` that will be partially decrypted
+        :param extended_base_hash: the extended base hash of the election that 
+                                   was used to generate t he ElGamal Ciphertext
+        :param nonce_seed: an optional value used to generate the `ChaumPedersenProof`
+                           if no value is provided, a random number will be used.
+        :param decrypt: an `AuxiliaryDecrypt` function to decrypt the missing guardina private key backup
+        :return: a `Tuple[ElementModP, ChaumPedersenProof]` of the decryption and its proof
+        """
+        if nonce_seed is None:
+            nonce_seed = int_to_q_unchecked(randbelow(Q))
+
+        backup = self._guardian_election_partial_key_backups.get(missing_guardian_id)
+        if backup is None:
+            log_warning(
+                f"compensate decrypt guardian {self.object_id} missing backup for {missing_guardian_id}"
+            )
+            return None
+
+        decrypted_value = decrypt(backup.encrypted_value, self._auxiliary_keys)
+        partial_secret_key = int_to_q_unchecked(int(decrypted_value))
+
+        # 𝑀_{𝑖,l} = 𝐴^P𝑖_{l}
+        partial_decryption = elgamal.partial_decrypt(partial_secret_key)
+
+        # 𝑀_{𝑖,l} = 𝐴^𝑠𝑖 mod 𝑝 and 𝐾𝑖 = 𝑔^𝑠𝑖 mod 𝑝
+        proof = make_chaum_pedersen(
+            message=elgamal,
```

Comment:
why are these are manually defined? for readability?

Reply:
yes in the cases of these proofs i wanted to call out the names of the parameters and how they map in this context.  these things are used all over the place and I found it difficult to follow when i was debugging.  (e.g. wait, which property is the 3rd ordered param? hang on let me go inspect the method). I think having this specified with named params here helps (me, at least) better understand what is happening